src/api/firebase.js

```diff
@@ -189,12 +189,32 @@ export async function addItem(listPath, { itemName, daysUntilNextPurchase }) {
 	});
 }
 
-export async function updateItem() {
-	/**
-	 * TODO: Fill this out so that it uses the correct Firestore function
-	 * to update an existing item. You'll need to figure out what arguments
-	 * this function must accept!
-	 */
+export async function updateItem(listPath, item) {
+	// creates a reference to the item collection
+	const listCollectionRef = collection(db, listPath, 'items');
+	// creates a reference to current item by passing the item's id
+	const itemRef = doc(listCollectionRef, item.id);
+
+	// updating the item in the item's collection by passing it the itemRef
+	await updateDoc(itemRef, {
```

Comment:
For me the updateItem function works as described but I am commenting here because it is unclear to me how this will function in practice. Do we not want the totalPurchases number to persist so the app can use that data to suggest the next time the user will want to purchase it again? Also to keep track of how often/much the user buys the item? I hope we can discuss this in further detail. I think I am straying outside the scope of the issue but I want to record my questions. What will be the purpose of having the totalPurchases property if it will only vacillate between 0 and 1? Should it be renamed then? Why is dateOfLastPurchase cleared instead of only updated each time the item is purchased? Why doesn't the app want to remember when it was last purchased even if it isnt checked? If the check is to indicate that the item was purchased, I do not see how unchecking it will remove the need to know when it was last purchased.

Reply:
Hey Paige!
My current understanding is that the user would be interacting with this list when they're at the store, checkin off as they shop. Then the app itself will uncheck the item after 24hrs (they don't need to manually do this).

So we thought, what if someone accidentally checks an item? Or they check an item only to realized they couldn't make the purchase?

If the app is designed to auto uncheck for the user, then we could assume if a user is unchecking it themselves, its because of one of the two reasons above.

If that occurs, then we thought we should deduct 1 from totalPurchases, and clear the dateLastPurchased timestamp. This would keep totalPurchases and dateLastPurchased accurate. 

This may help to answer your question of totalPurchases vacillating between 0 and 1, because you were probably clicking the checkbox on and off.

Try checking a item off, manually resetting the time, recording the totalPurchases and repeating this action a few times. You should see the totalPurchases increment. The manually uncheck, it should decrease by 1 to update the discrepancy. 

I hope this explains our current logic, and I hope that our interpretation of the user interacting with this list is right for the issue.

We attempted to explain this with a couple of `<p>` tags, but it was tricky to explain.

Reply:
Okay so the totalPurchases only decreases by one by physically/manually unchecking the box. Thank you, this helps!!

Reply:
Yeah!

At one point we had it so when an item was checked off, it only relied on the 24hr reset to uncheck. This version disabled the user from being able to uncheck an item manually. 

But we thought, if we were a user that would be really annoying, not being able to physically/manually uncheck, especially if you didn't mean to check it.

So I guess, we need some better instructions above the list to explain the way the user should interact with the checkbox functionality. 

*provided we are on the right track with how we are interpreting the functionality